matter_persistence/redis/manager.py

```diff
@@ -211,10 +212,21 @@ async def save_many_with_keys(
         :param values_to_store: a dictionary, mapping keys to values (see note above)
         :param object_class: optional model to facilitate serialisation of data
         :param expiration_in_seconds: cache expiration time in seconds
+        :param use_key_as_is: whether to use key as is
         """
         object_name = object_class.__name__ if object_class else None
 
-        async with self.__get_cache_client(for_writing=True) as cache_client:
+        if use_key_as_is:
```

Comment:
Putting this `if` here causes the code to side-step my logic of storing sequences of objects. It would be better if you moved this `if use_key_as_is` to inside the existing `ifs`, only affecting the key generation.